### PR TITLE
support docker build-arg

### LINKED ISSUE #55: put should support build-arg
Docker 1.9 introduced [build-arg](https://docs.docker.com/engine/reference/builder/#/arg), it could be nice to support this feature. 

Something like this (similar to ca-certs)
build-arg: Optional. An array of objects with the following format:
```yaml
build-arg:
- key: maintainer
  value: vito@concourse.org
- key: ruby-gem-mirror
  value: https://my-mirror.com/
```

### PR DESCRIPTION
**Pull request by o-orand**
2 ways to set:
 * using param in put
 * using a yaml file

When both set, the result contains all build arguments (may have duplicate!).

close #55

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -143,6 +143,21 @@ version is the image's digest.
...
 
+* `build_args`: *Optional.*  Default ``. List of docker build arguments. Format: `<key>: <value>`.
+
+    ```yaml
+      build_args:
+        argInt: "1"
+        argBool: "true"
+        argEmail: me@yopmail.com
+    ```            
+* `build_args_file`: *Optional.*  Default ``. A Yaml filename containing docker build arguments.
```
Comment (vito) [inline README.md]: JSON, not YAML, right?

File: assets/out [hunk 2]
```diff
@@ -100,10 +102,19 @@ elif [ -n "$build" ]; then
     docker_pull "${repository}:${cache_tag}"
   fi
 
+  expanded_build_args=""
+  if [ -n "$build_args" ]; then
+    expanded_build_args=$(echo $build_args|jq -r 'with_entries(.key |= "--build-arg " + . )|with_entries(.key = .key + "=" +.value)|keys|join(" ")')
```
Comment (vito) [inline assets/out]: I'm a bit worried about escaping semantics here (and in the call site), e.g. if the value has characters like `!` or spaces. Have you tried that out?

Comment (o-orand) [inline assets/out]: You're right, this doesn't support `space` neither `!` :disappointed: . I'm going to see how to escape special characters. I'm not sure it's possible using jq.

File: README.md (hunk 1 above)
Comment (o-orand) [inline README.md]: right, file is parsed with jq...

File: assets/out (hunk 2 above)
Comment (vito) [inline assets/out]: To be honest it's probably time to rewrite `out` in Go; it's been getting pretty complicated over time. I'm not sure if you wanna bite that bullet as part of this PR though. We've at least done that for `check` already.

Comment (o-orand) [inline assets/out]: Indeed, it could be nice to rewrite it in Go, but I'm not ready to do it yet. Maybe later :smile:

**Comment by vito** [comment]
I'll pull this in as-is even though we know it'll break with fancier arguments. Thanks!

I'll update the README to say the file must be JSON, not YAML.

Paging Dr. @zachgersh for the Go rewrite. 😉

**Comment by zachgersh** [comment]
@vito page received but you picked git first 👯

**Comment by vito** [comment]
did not - docker image resource is way more important. i said git over bosh deployment resource :)

**Pull request merged**